### DISCUSSION TITLE
How to continue using same trace from main thread in worker thread with Auto Instrumentation

### DISCUSSION THREAD

**Discussion by suman2007** [open-telemetry/opentelemetry-java-instrumentation#10796] [Q&A]
I have main thread(service thread) and creating 2 separate threads Thread-1 and Thread-2 and trying to send the rest call from both thread, but there is not any Trace Id found from two thread.
Expectations: Trace Id must be same in Thread-1 and Thread-2 which was generated in Main Thread.

![image](https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/55128431/8c1710c8-ca20-4dab-bb3a-aa8e0ea910b4)

**Comment by laurit** [comment]
https://opentelemetry.io/docs/languages/java/instrumentation/#context-propagation-between-threads

**Reply by suman2007** [reply]
These manual changes are required in every portion from where a new thread spawned although I am using Autointrumentation.

**Reply by laurit** [reply]
Auto instrumentation propagates context when using executors listed in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/d852eade109f4eefa92ff78dcc43b65acbf77a6e/instrumentation/executors/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/executors/ExecutorMatchers.java#L47-L99 When something else is used manual instrumentation may be needed.

**Reply by suman2007** [reply]
Thanks for the update!!

**Reply by wdengw** [reply]
> https://opentelemetry.io/docs/languages/java/instrumentation/#context-propagation-between-threads

@laurit The Context Propagation Between Threads anchor doesn't exists on the current page: https://opentelemetry.io/docs/languages/java/instrumentation.  Would you share the updated link?

**Comment by suman2007** [comment]
Is there any reason that Context Propagation in Auto instrumentation is not supported for the traditional Thread approach or there is plan to incorporate this missing thing in the upcoming release?

**Reply by trask** [reply]
Check out https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/2527
